streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -748,24 +702,27 @@ private void assignTasksToClients(final Set<String> allSourceTopics,
         final boolean lagComputationSuccessful =
             populateClientStatesMap(clientStates, clientMetadataMap, taskForPartition, changelogsByStatefulTask);
 
-        // assign tasks to clients
         final Set<TaskId> allTasks = partitionsForTask.keySet();
-        final Set<TaskId> standbyTasks = changelogsByStatefulTask.keySet();
-
-        if (lagComputationSuccessful) {
-            final Map<TaskId, SortedSet<RankedClient<UUID>>> statefulTasksToRankedCandidates =
-                buildClientRankingsByTask(standbyTasks, clientStates, acceptableRecoveryLag());
-            log.trace("Computed statefulTasksToRankedCandidates map as {}", statefulTasksToRankedCandidates);
-        }
+        final Set<TaskId> statefulTasks = changelogsByStatefulTask.keySet();
 
         log.debug("Assigning tasks {} to clients {} with number of replicas {}",
             allTasks, clientStates, numStandbyReplicas());
 
-        final StickyTaskAssignor<UUID> taskAssignor = new StickyTaskAssignor<>(clientStates, allTasks, standbyTasks);
-        if (!lagComputationSuccessful) {
-            taskAssignor.preservePreviousTaskAssignment();
+        final TaskAssignor<UUID> taskAssignor;
+        if (highAvailabilityEnabled) {
+            if (lagComputationSuccessful) {
+                taskAssignor = new HighAvailabilityTaskAssignor<>(
+                    clientStates,
+                    allTasks,
+                    statefulTasks,
+                    assignmentConfigs);
+            } else {
+                taskAssignor = new StickyTaskAssignor<>(clientStates, allTasks, statefulTasks, assignmentConfigs, true);
+            }
+        } else {
+            taskAssignor = new StickyTaskAssignor<>(clientStates, allTasks, statefulTasks, assignmentConfigs, false);
```

Comment:
```suggestion
            taskAssignor = new StickyTaskAssignor<>(clientStates, allTasks, statefulTasks, assignmentConfigs, !lagComputationSuccessful);
```

It still seems like this is necessary to preserve the agreed on approach to https://github.com/ableegoldman/kafka/pull/2#discussion_r394736803

Reply:
Ok, I talked with Sophie offline, and I was mistaken about the impact of `lagComputationSuccessful`. I was thinking that `StickyTaskAssignor` couldn't produce a valid assignment without the lag computation, but this isn't true. So I'm resolving this conversation.